lib/query.js

```diff
@@ -4924,6 +4924,13 @@ Query.prototype.exec = function exec(op, callback) {
   const _this = this;
   // Ensure that `exec()` is the first thing that shows up in
   // the stack when cast errors happen.
+  const execOptions = this._optionsForExec(this.model);
+  if (execOptions.sort) {
```

Comment:
Let's just use `this.options.sort`. `_optionsForExec()` clones the object, so this would incur a performance penalty.